UKHomeOffice/sgmr-service
# 2.0 SPCR-254 New add person form

## Description

- setup new people section in pages
- add new people multipage form
- move person form validation to people section
- make v2 component folder 
- make FormFieldError component for v2 page needs
- make v2 apiHooks for v2 page needs
- point links to new people form

NOT IN THIS PR
- editing people

To test

- go to /people
- click on save new person
- click on continue
- > all fields should error
- fill in date of birth with a date in the future
- click continue
- > date of birth field should error as invalid date
- fill in date of birth with an invalid date (e.g. month=22)
- > date of birth field should error as invalid date
- fill in other fields
- click continue
- > you should be taken to page 2 of the form
- go back to page 1
- > data should be persisted in the fields
- click continue
- > you should be taken to page 2 of the form
- click submit
- > all fields should error
- fill in an invalid expiry date (e.g. month=22)
- click submit
- > expiry date should error as invalid date
- enter an expiry date in the past
- > expiry date should error as an invalid date
- select document type of `Another travel document`
- > a free text field should be displayed
- click submit 
- > document type should show error
- enter text in the please specify field
- click submit
- > document type should NOT error
- change document type to passport
- enter document issuing state in lowercase letters (e.g. gbr)
- click submit
- > document issuing state should not error
- fill in the rest of the fields
- click submit
- > you should be returned to the people page with the list of people
- > your person should be visible in that list (in alphabetical order surname>firstname)

## Developer Checklist

\* Required






Ensure you delete the branch once the Pull Request is merged.
